packages/opentelemetry-instrumentation-langchain/opentelemetry/instrumentation/langchain/callback_wrapper.py

```diff
@@ -21,81 +22,136 @@ def default(self, o: Any) -> str:
             return str(o)
 
 
-def get_name(to_wrap, instance) -> str:
-    return f"{instance.get_name()}.langchain.{to_wrap.get('kind')}"
-
-
-def get_kind(to_wrap) -> str:
-    return to_wrap.get("kind")
-
-
 @_with_tracer_wrapper
 def callback_wrapper(tracer, to_wrap, wrapped, instance, args, kwargs):
+    """Hook into the invoke function, config is part of args, 2nd place.
+    sources:
+    https://python.langchain.com/v0.2/docs/how_to/callbacks_attach/
+    https://python.langchain.com/v0.2/docs/how_to/callbacks_runtime/
+    """
     if context_api.get_value(_SUPPRESS_INSTRUMENTATION_KEY):
         return wrapped(*args, **kwargs)
-    kind = get_kind(to_wrap)
-    name = get_name(to_wrap, instance)
-    cb = SyncSpanCallbackHandler(tracer, name, kind)
-    if "callbacks" in kwargs:
-        if not any(isinstance(c, SyncSpanCallbackHandler) for c in kwargs["callbacks"]):
-            # Avoid adding the same callback twice, e.g. SequentialChain is also a Chain
-            kwargs["callbacks"].append(cb)
+    kind = to_wrap.get("kind")
```

Comment:
One thing I would here is extract all this logic (except for the actual call to `wrapped` to a separate method so you can decorate it with `@dont_throw` (and so for the callback methods below)